File: Barone.api/Models/PedidoModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Barone.api.Models
{
    public class PedidoModel
    {
        public long id { set; get; }
        public int IdCliente { set; get; }
        public DateTime fechaPedido { set; get; }
        public DateTime fechaPactada { set; get; }
        public string DetallePedido { set; get; }
        public string TotalBarriles { set; get; }
        public string Estado { set; get; }
        public long? idEntrega { set; get; }
        public ClientesModel Cliente { set; get; }
        public MovimientosModel Entrega { set; get; }
    }
}

File: Barone.Client/src/app/modules/shared/models/pedido.model.ts
import { ClientsModel } from "./clients.model";
import { Constructable } from "src/app/core/models/constructable";
import { RowEntregaLata } from "./row-entrega";

export class PedidoModel extends Constructable<Partial<PedidoModel>> {
    id: number;
    public IdCliente: number;
    fechaPedido: string;
    fechaPactada: string;
    Cliente: ClientsModel;
    DetallePedido: string;
    TotalBarriles: number;
    TotalLatas:number;
    Estado: string;
    idEntrega: number;
    DetallePedidoProducto:string;
    DetalleProducto: RowEntregaLata[];
}
